nemo/core/connectors/save_restore_connector.py

```diff
@@ -479,37 +481,35 @@ def _handle_artifacts(self, model, nemo_file_folder):
             # Need to step into nemo archive to extract file
             # Get path where the command is executed - the artifacts will be "retrieved" there
             # (original .nemo behavior)
-            cwd = os.getcwd()
-            # Step into the nemo archive to try and find the file
-            # TemporaryDirectory context must always be outer to try-catch chdir otherwise it crashes on Windows
-            with tempfile.TemporaryDirectory() as archive_dir:
-                try:
-                    # unpack all restorations paths (nemo checkpoints)
-                    # in nemo checkpoints all resources contain hash in name, so there should be no collisions
-                    for path in restoration_paths:
```

Comment:
Continue with old logic, wherever extraction is required, only change that part to extract filtered filer rather than all checkpoint files.

Reply:
What is "old logic" here? I don't see the reason to create a tmp dir (potentially on a diff file system) just to pull one file into the nemo dir. In my experiments with a large model, b/c this work was only done on rank=0, it took >10min and then NCCL timed out.

I believe this is a general improvement, but also helps with a functional issue I am facing.

Reply:
Creation of temp dir is not the slowdown, it's the extraction of the entire weight file that's slow. You can keep the temporary directory extraction, and still extract just one tokenizer file or yaml file to it just as always, which will be a non issue in time. You do not need to modify the current code in it's entirety to achieve your primary goal of extracting particular file type without full extraction of tarfiles to any kind of directory. That is what I meant.

Reply:
> You do not need to modify the current code in it's entirety to achieve your primary goal of extracting particular file type without full extraction of tarfiles to any kind of directory.

If I copy to a temp dir and then copy to the final tmp nemo file dir, I'll copy twice. Are you proposing keeping two copies instead of this refactor which should only copy once b/c it reduces the diff?